app/src/main/java/com/sun/basic_japanese/flashcard/allpages/FlashCardFragment.kt

```diff
@@ -0,0 +1,75 @@
+package com.sun.basic_japanese.flashcard.allpages
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import com.sun.basic_japanese.R
+import com.sun.basic_japanese.base.BaseFragment
+import com.sun.basic_japanese.constants.BasicJapaneseConstants.EMPTY_STRING
+import com.sun.basic_japanese.constants.BasicJapaneseConstants.FLASHCARD
+import com.sun.basic_japanese.data.model.Alphabet
+import com.sun.basic_japanese.data.model.FlashCardMessage
+import com.sun.basic_japanese.data.model.FlashCardsMessage
+import com.sun.basic_japanese.flashcard.eachpage.FlashCardPageFragment
+import com.sun.basic_japanese.flashcard.adapter.FlashCardPagerAdapter
+import kotlinx.android.synthetic.main.fragment_flash_card.*
+
+class FlashCardFragment : BaseFragment(), FlashCardContract.View {
+
+    private var flashCardPresenter: FlashCardContract.Presenter? = null
+    private var currentPosition = START_POSITION
+    private var alphabetItems = listOf<Alphabet?>()
+    private var type = EMPTY_STRING
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.run {
```

Comment:
Đừng nhét nhiều thứ vào scope function vậy vì sẽ khiến code khó đọc.
```suggestion
        val flashCards = arguments?.getParcelable<FlashCardsMessage>(FLASHCARD)
        flashCards?.let {
            ...
```